### ISSUE TITLE
Point datatype

### ISSUE THREAD

**Issue by zmactep** [boltprotocol/boltprotocol#7]
Neo4j specification [points](https://neo4j.com/docs/developer-manual/current/cypher/functions/spatial/#functions-distance) to Point datatype (both 2D and 3D), but there is no information about it in BOLT Protocol specification. Should we add it?

**Comment by cleishm** [comment]
Yes, perhaps a call should be put out proposing to standardize a bolt v2?

**Comment by zmactep** [comment]
Yes, as I see this as an error:
`Point is not supported as a return type in Bolt protocol version 1. Please make sure driver supports at least protocol version 2. Driver upgrade is most likely required.`

**Comment by cleishm** [comment]
Yes, neo has implemented their own v2, but there is no standard as yet. It would be good to get implementors together to specify one.

**Comment by zmactep** [comment]
I am working for new neo4j protocol support in my Haskell driver (zmactep/hasbolt). After that I can commit some specs for v2 new data types support.

**Comment by cleishm** [comment]
Cool. We might like to reach out to other server implementors that use bolt, and invite them to discuss.

**Comment by zmactep** [comment]
Are there any? I know only neo, so I'd like to meet some bolt-based servers maintainers.

**Comment by cleishm** [comment]
I believe there are others, such as MemGraph.

Also, I should have included driver implementors in that statement, as they’ll also have valuable input.

**Comment by zmactep** [comment]
So, all new types are just records. Neo doesn't specify this anywhere, but I got it from sources. Here is the table:
* Point2D
```
signature = 'X'
fields = { crs :: CoordinateReferenceSystem (see below)
         , x   :: Double
         , y   :: Double
         }
```
* Point3D
```
signature = 'Y'
fields = { crs :: CoordinateReferenceSystem
         , x   :: Double
         , y   :: Double
         , z   :: Double
         }
```
* Duration
```
signature = 'E'
fields = { months  :: Integer
         , days    :: Integer
         , seconds :: Integer
         , nanos   :: Integer
         }
```
* Date
```
signature = 'D'
fields = { epochDays :: Integer
         }
```
* Time
```
signature = 'T'
fields = { nanosOfDayLocal :: Integer
         , offsetSeconds   :: Integer
         }
```
* LocalTime
```
signature = 't'
fields = { nanosOfDay :: Integer
         }
```
* LocalDateTime
```
signature = 'd'
fields = { epochSeconds :: Integer
         , nano         :: Integer
         }
```
* DateTimeWithZoneOffset
```
signature = 'F'
fields = { epochSecondsLocal :: Integer
         , nano              :: Integer
         , offsetSeconds     :: Integer
         }
```
* DateTimeWithZoneName
```
signature = 'f'
fields = { epochSecondsLocal :: Integer
         , nano              :: Integer
         , zoneId            :: Integer
         }
```

Codes of Coordinate Reference Systems:
* Cartesian 2D — 7203
* Cartesian 3D — 9157
* WGS-84 2D — 4326
* WGS-84 3D — 4979
